src/qutip_cupy/__init__.py

```diff
@@ -1 +1,25 @@
 from .version import version as __version__
+from .cupy_dense import *
+
+
+from qutip.core import data
```

Comment:
Put all imports together at the top of the file (unless they're inside functions or methods, e.g. as an optional dependency).

For extra credit, divide the imports into three paragraphs -- one for modules in the standard library, one for third party modules, and one for modules from this package.